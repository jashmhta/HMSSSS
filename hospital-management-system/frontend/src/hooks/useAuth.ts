import { useState, useEffect } from 'react';

interface User {
  id: string;
  email: string;
  firstName: string;
  lastName: string;
  role: string;
}

interface AuthState {
  user: User | null;
  isAuthenticated: boolean;
  isLoading: boolean;
}

interface UseAuthReturn {
  user: User | null;
  isAuthenticated: boolean;
  isLoading: boolean;
  login: (credentials: { email: string; password: string; rememberMe?: boolean }) => Promise<void>;
  logout: () => void;
  register: (userData: { email: string; password: string; firstName: string; lastName: string }) => Promise<void>;
  updateUser: (userData: Partial<User>) => Promise<void>;
}

export const useAuth = (): UseAuthReturn => {
  const [state, setState] = useState<AuthState>({
    user: null,
    isAuthenticated: false,
    isLoading: false,
  });

  const login = async (credentials: { email: string; password: string; rememberMe?: boolean }) => {
    setState(prev => ({ ...prev, isLoading: true }));

    try {
      // Simulate API call
      await new Promise(resolve => setTimeout(resolve, 1000));

      // Mock successful login
      const user: User = {
        id: '1',
        email: credentials.email,
        firstName: 'Test',
        lastName: 'User',
        role: credentials.email.includes('admin') ? 'admin' : 'user',
      };

      localStorage.setItem('token', 'mock-jwt-token');
      localStorage.setItem('userRole', user.role);

      setState({
        user,
        isAuthenticated: true,
        isLoading: false,
      });
    } catch (error) {
      setState(prev => ({ ...prev, isLoading: false }));
      throw error;
    }
  };

  const logout = () => {
    localStorage.removeItem('token');
    localStorage.removeItem('userRole');
    setState({
      user: null,
      isAuthenticated: false,
      isLoading: false,
    });
  };

  const register = async (userData: { email: string; password: string; firstName: string; lastName: string }) => {
    setState(prev => ({ ...prev, isLoading: true }));

    try {
      // Simulate API call
      await new Promise(resolve => setTimeout(resolve, 1000));

      const user: User = {
        id: '1',
        email: userData.email,
        firstName: userData.firstName,
        lastName: userData.lastName,
        role: 'user',
      };

      setState({
        user,
        isAuthenticated: true,
        isLoading: false,
      });
    } catch (error) {
      setState(prev => ({ ...prev, isLoading: false }));
      throw error;
    }
  };

  const updateUser = async (userData: Partial<User>) => {
    setState(prev => ({ ...prev, isLoading: true }));

    try {
      // Simulate API call
      await new Promise(resolve => setTimeout(resolve, 500));

      setState(prev => ({
        ...prev,
        user: prev.user ? { ...prev.user, ...userData } : null,
        isLoading: false,
      }));
    } catch (error) {
      setState(prev => ({ ...prev, isLoading: false }));
      throw error;
    }
  };

  useEffect(() => {
    // Check for existing token on mount
    const token = localStorage.getItem('token');
    if (token) {
      const userRole = localStorage.getItem('userRole');
      setState({
        user: {
          id: '1',
          email: 'user@example.com',
          firstName: 'Test',
          lastName: 'User',
          role: userRole || 'user',
        },
        isAuthenticated: true,
        isLoading: false,
      });
    }
  }, []);

  return {
    ...state,
    login,
    logout,
    register,
    updateUser,
  };
};