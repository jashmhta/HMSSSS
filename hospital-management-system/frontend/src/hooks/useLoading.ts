import { useState, useCallback, useEffect } from 'react';

export interface ILoadingState {
  isLoading: boolean;
  progress?: number;
  message?: string;
  error?: string | null;
}

export interface IUseLoadingOptions {
  initialMessage?: string;
  timeout?: number;
  onError?: (error: Error) => void;
  onSuccess?: () => void;
}

export interface ILoadingTask<T = any> {
  id: string;
  name: string;
  status: 'pending' | 'loading' | 'completed' | 'failed';
  progress?: number;
  message?: string;
  error?: string;
  result?: T;
  startTime?: number;
  endTime?: number;
}

export function useLoading(options: IUseLoadingOptions = {}) {
  const [loadingState, setLoadingState] = useState<ILoadingState>({
    isLoading: false,
    ...(options.initialMessage && { message: options.initialMessage }),
    error: null,
  });

  const [tasks, setTasks] = useState<ILoadingTask[]>([]);

  const startLoading = useCallback((message?: string) => {
    setLoadingState({
      isLoading: true,
      message: message || options.initialMessage || 'Loading...',
      error: null,
    });
  }, [options.initialMessage]);

  const stopLoading = useCallback(() => {
    setLoadingState(prev => ({
      ...prev,
      isLoading: false,
    }));
    options.onSuccess?.();
  }, [options.onSuccess]);

  const setProgress = useCallback((progress: number, message?: string) => {
    setLoadingState(prev => ({
      ...prev,
      progress,
      ...(message || prev.message ? { message: message || prev.message } : {}),
    }));
  }, []);

  const setError = useCallback((error: Error | string) => {
    const errorMessage = error instanceof Error ? error.message : error;
    setLoadingState(prev => ({
      ...prev,
      isLoading: false,
      error: errorMessage,
    }));
    options.onError?.(error instanceof Error ? error : new Error(errorMessage));
  }, [options.onError]);

  const reset = useCallback(() => {
    setLoadingState({
      isLoading: false,
      ...(options.initialMessage && { message: options.initialMessage }),
      error: null,
    });
  }, [options.initialMessage]);

  // Task management
  const addTask = useCallback(<T = any>(name: string, taskFn: () => Promise<T>): Promise<T> => {
    const taskId = Math.random().toString(36).substr(2, 9);
    const newTask: ILoadingTask<T> = {
      id: taskId,
      name,
      status: 'pending',
      startTime: Date.now(),
    };

    setTasks(prev => [...prev, newTask]);

    const updateTask = (updates: Partial<ILoadingTask<T>>) => {
      setTasks(prev => prev.map(task =>
        task.id === taskId ? { ...task, ...updates } : task,
      ));
    };

    return new Promise((resolve, reject) => {
      updateTask({ status: 'loading' });
      startLoading(name);

      taskFn()
        .then((result) => {
          updateTask({
            status: 'completed',
            result,
            endTime: Date.now(),
          });
          stopLoading();
          resolve(result);
        })
        .catch((error) => {
          updateTask({
            status: 'failed',
            error: error.message,
            endTime: Date.now(),
          });
          setError(error);
          reject(error);
        });
    });
  }, [startLoading, stopLoading, setError]);

  const removeTask = useCallback((taskId: string) => {
    setTasks(prev => prev.filter(task => task.id !== taskId));
  }, []);

  const clearTasks = useCallback(() => {
    setTasks([]);
  }, []);

  const getTaskStats = useCallback(() => {
    const total = tasks.length;
    const completed = tasks.filter(t => t.status === 'completed').length;
    const failed = tasks.filter(t => t.status === 'failed').length;
    const inProgress = tasks.filter(t => t.status === 'loading').length;

    return {
      total,
      completed,
      failed,
      inProgress,
      successRate: total > 0 ? (completed / total) * 100 : 0,
    };
  }, [tasks]);

  // Auto-cleanup on timeout
  useEffect(() => {
    if (loadingState.isLoading && options.timeout) {
      const timer = setTimeout(() => {
        if (loadingState.isLoading) {
          setError(new Error(`Loading timeout after ${options.timeout}ms`));
        }
      }, options.timeout);

      return () => clearTimeout(timer);
    }
    return () => {}; // Return empty cleanup function when no timer is set
  }, [loadingState.isLoading, options.timeout, setError]);

  return {
    loadingState,
    tasks,
    startLoading,
    stopLoading,
    setProgress,
    setError,
    reset,
    addTask,
    removeTask,
    clearTasks,
    getTaskStats,
  };
}

// Specialized hooks for common loading patterns
export function useAsyncData<T>(
  asyncFn: () => Promise<T>,
  options: IUseLoadingOptions & { immediate?: boolean; deps?: any[] } = {},
) {
  const {
    immediate = true,
    deps = [],
    onError,
    onSuccess,
    ...loadingOptions
  } = options;

  const { loadingState, addTask } = useLoading({
    ...loadingOptions,
    ...(onError && { onError }),
    ...(onSuccess && { onSuccess }),
  });

  const [data, setData] = useState<T | null>(null);

  const execute = useCallback(async () => {
    try {
      const result = await addTask('Data fetch', asyncFn);
      setData(result);
      return result;
    } catch (error) {
      setData(null);
      throw error;
    }
  }, [asyncFn, addTask]);

  useEffect(() => {
    if (immediate) {
      execute().catch(console.error);
    }
  }, [immediate, execute, ...deps]);

  return {
    data,
    loadingState,
    execute,
    refetch: execute,
  };
}

export function useFormSubmission<T>(
  submitFn: (data: T) => Promise<any>,
  options: IUseLoadingOptions & { resetOnSuccess?: boolean } = {},
) {
  const { resetOnSuccess = false, ...loadingOptions } = options;
  const { loadingState, setError, reset } = useLoading(loadingOptions);

  const submit = useCallback(async (data: T) => {
    try {
      await submitFn(data);
      if (resetOnSuccess) {
        reset();
      }
    } catch (error) {
      setError(error as Error | string);
      throw error;
    }
  }, [submitFn, setError, reset, resetOnSuccess]);

  return {
    loadingState,
    submit,
    setError,
    reset,
  };
}

export function useInfiniteScroll<T>(
  loadMoreFn: (page: number) => Promise<{ items: T[]; hasMore: boolean }>,
  options: IUseLoadingOptions = {},
) {
  const { loadingState, setError, addTask } = useLoading(options);
  const [items, setItems] = useState<T[]>([]);
  const [page, setPage] = useState(1);
  const [hasMore, setHasMore] = useState(true);

  const loadMore = useCallback(async () => {
    if (!hasMore || loadingState.isLoading) return;

    try {
      const result = await addTask(`Load page ${page}`, () => loadMoreFn(page));
      setItems(prev => [...prev, ...result.items]);
      setHasMore(result.hasMore);
      setPage(prev => prev + 1);
    } catch (error) {
      setError(error as Error | string);
    }
  }, [page, hasMore, loadingState.isLoading, loadMoreFn, addTask, setError]);

  const reset = useCallback(() => {
    setItems([]);
    setPage(1);
    setHasMore(true);
  }, []);

  return {
    items,
    loadingState,
    hasMore,
    loadMore,
    reset,
  };
}

export function useRealtimeStatus<T>(
  getStatusFn: () => Promise<T>,
  options: IUseLoadingOptions & { interval?: number } = {},
) {
  const { interval = 5000, ...loadingOptions } = options;
  const { loadingState, setError, addTask } = useLoading(loadingOptions);
  const [status, setStatus] = useState<T | null>(null);

  useEffect(() => {
    const updateStatus = async () => {
      try {
        const result = await addTask('Status check', getStatusFn);
        setStatus(result);
      } catch (error) {
        setError(error as Error | string);
      }
    };

    updateStatus();
    const intervalId = setInterval(updateStatus, interval);

    return () => clearInterval(intervalId);
  }, [getStatusFn, interval, addTask, setError]);

  return {
    status,
    loadingState,
  };
}

// Smart loading utilities
export const loadingUtils = {
  getEstimatedTime: (tasks: ILoadingTask[]): number => {
    const completedTasks = tasks.filter(t => t.status === 'completed');
    if (completedTasks.length === 0) return 0;

    const totalTime = completedTasks.reduce((sum, task) => {
      return sum + ((task.endTime || 0) - (task.startTime || 0));
    }, 0);

    return totalTime / completedTasks.length;
  },

  getSmartMessage: (state: ILoadingState, tasks: ILoadingTask[]): string => {
    if (state.error) return `Error: ${state.error}`;
    if (state.message) return state.message;

    const activeTasks = tasks.filter(t => t.status === 'loading');
    if (activeTasks.length === 0) return 'Loading...';

    if (activeTasks.length === 1) {
      return activeTasks[0]?.name || 'Loading...';
    }

    return `${activeTasks.length} tasks in progress...`;
  },

  calculateProgress: (tasks: ILoadingTask[]): number => {
    if (tasks.length === 0) return 0;

    const completedTasks = tasks.filter(t => t.status === 'completed').length;
    return Math.round((completedTasks / tasks.length) * 100);
  },

  shouldShowSkeleton: (state: ILoadingState, delay: number = 300): boolean => {
    return state.isLoading && Date.now() - (state.progress || 0) > delay;
  },
};