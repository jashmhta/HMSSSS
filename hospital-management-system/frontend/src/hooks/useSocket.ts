import { useEffect, useRef } from 'react';
import { io, Socket } from 'socket.io-client';

const useSocket = () => {
  const socketRef = useRef<Socket | null>(null);

  useEffect(() => {
    // Connect to the backend socket server
    socketRef.current = io(process.env['NEXT_PUBLIC_BACKEND_URL'] || 'http://localhost:3001', {
      transports: ['websocket', 'polling'],
    });

    // Cleanup on unmount
    return () => {
      if (socketRef.current) {
        socketRef.current.disconnect();
      }
    };
  }, []);

  const emit = (event: string, data?: any) => {
    if (socketRef.current) {
      socketRef.current.emit(event, data);
    }
  };

  const on = (event: string, callback: (...args: any[]) => void) => {
    if (socketRef.current) {
      socketRef.current.on(event, callback);
    }
  };

  const off = (event: string, callback?: (...args: any[]) => void) => {
    if (socketRef.current) {
      if (callback) {
        socketRef.current.off(event, callback);
      } else {
        socketRef.current.off(event);
      }
    }
  };

  return { socket: socketRef.current, emit, on, off };
};

export default useSocket;