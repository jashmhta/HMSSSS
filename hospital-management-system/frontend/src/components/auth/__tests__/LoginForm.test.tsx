import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import LoginForm from '../LoginForm';

// Mock the useAuth hook
jest.mock('../../../hooks/useAuth');
import { useAuth } from '../../../hooks/useAuth';

const mockUseAuth = useAuth as jest.MockedFunction<typeof useAuth>;

describe('LoginForm', () => {
  const mockLogin = jest.fn();

  beforeEach(() => {
    mockUseAuth.mockReturnValue({
      login: mockLogin,
      user: null,
      isAuthenticated: false,
      isLoading: false,
      logout: jest.fn(),
      register: jest.fn(),
      updateUser: jest.fn(),
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders login form correctly', () => {
    render(<LoginForm />);

    expect(screen.getByText(/sign in to your account/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/email address/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/password/i)).toBeInTheDocument();
    expect(screen.getByTestId('submit-button')).toBeInTheDocument();
    expect(screen.getByText(/forgot your password\?/i)).toBeInTheDocument();
    expect(screen.getByText(/don't have an account\?/i)).toBeInTheDocument();
  });

  it('allows user to type in email and password fields', async () => {
    const user = userEvent.setup();
    render(<LoginForm />);

    const emailInput = screen.getByLabelText(/email/i);
    const passwordInput = screen.getByPlaceholderText(/password/i);

    await user.type(emailInput, 'test@example.com');
    await user.type(passwordInput, 'password123');

    expect(emailInput).toHaveValue('test@example.com');
    expect(passwordInput).toHaveValue('password123');
  });

  it('validates email format', async () => {
    const user = userEvent.setup();
    render(<LoginForm />);

    const emailInput = screen.getByPlaceholderText(/email address/i);
    const submitButton = screen.getByTestId('submit-button');

    await user.type(emailInput, 'invalid-email');
    await user.click(submitButton);

    await waitFor(() => {
      expect(screen.getByText(/please enter a valid email address/i)).toBeInTheDocument();
    });

    expect(mockLogin).not.toHaveBeenCalled();
  });

  it('validates required fields', async () => {
    const user = userEvent.setup();
    render(<LoginForm />);

    const submitButton = screen.getByTestId('submit-button');
    await user.click(submitButton);

    await waitFor(() => {
      expect(screen.getByText(/email is required/i)).toBeInTheDocument();
      expect(screen.getByText(/password is required/i)).toBeInTheDocument();
    });

    expect(mockLogin).not.toHaveBeenCalled();
  });

  it('calls login function with correct credentials', async () => {
    const user = userEvent.setup();
    mockLogin.mockResolvedValue({ success: true });

    render(<LoginForm />);

    const emailInput = screen.getByPlaceholderText(/email address/i);
    const passwordInput = screen.getByPlaceholderText(/password/i);
    const submitButton = screen.getByTestId('submit-button');

    await user.type(emailInput, 'test@example.com');
    await user.type(passwordInput, 'password123');
    await user.click(submitButton);

    await waitFor(() => {
      expect(mockLogin).toHaveBeenCalledWith({
        email: 'test@example.com',
        password: 'password123',
      });
    });
  });

  it('shows loading state during login', async () => {
    const user = userEvent.setup();
    mockLogin.mockReturnValue(new Promise(() => {})); // Never resolves

    render(<LoginForm />);

    const emailInput = screen.getByPlaceholderText(/email address/i);
    const passwordInput = screen.getByPlaceholderText(/password/i);
    const submitButton = screen.getByTestId('submit-button');

    await user.type(emailInput, 'test@example.com');
    await user.type(passwordInput, 'password123');
    await user.click(submitButton);

    expect(submitButton).toBeDisabled();
    expect(screen.getByText(/signing in/i)).toBeInTheDocument();
  });

  it('shows error message when login fails', async () => {
    const user = userEvent.setup();
    mockLogin.mockRejectedValue(new Error('Invalid credentials'));

    render(<LoginForm />);

    const emailInput = screen.getByPlaceholderText(/email address/i);
    const passwordInput = screen.getByPlaceholderText(/password/i);
    const submitButton = screen.getByTestId('submit-button');

    await user.type(emailInput, 'test@example.com');
    await user.type(passwordInput, 'wrongpassword');
    await user.click(submitButton);

    await waitFor(() => {
      expect(screen.getByText(/invalid credentials/i)).toBeInTheDocument();
    });
  });

  it('remembers me checkbox works', async () => {
    const user = userEvent.setup();
    render(<LoginForm />);

    const rememberMeCheckbox = screen.getByRole('checkbox', { name: /remember me/i });

    expect(rememberMeCheckbox).not.toBeChecked();

    await user.click(rememberMeCheckbox);

    expect(rememberMeCheckbox).toBeChecked();
  });

  it('navigates to register page when register link is clicked', async () => {
    const user = userEvent.setup();
    render(<LoginForm />);

    const registerLink = screen.getByRole('button', { name: /sign up/i });

    await user.click(registerLink);

    expect(window.location.pathname).toBe('/register');
  });

  it('navigates to forgot password page when forgot password link is clicked', async () => {
    const user = userEvent.setup();
    render(<LoginForm />);

    const forgotPasswordLink = screen.getByRole('button', { name: /forgot password/i });

    await user.click(forgotPasswordLink);

    expect(window.location.pathname).toBe('/forgot-password');
  });

  it('shows password visibility toggle', async () => {
    const user = userEvent.setup();
    render(<LoginForm />);

    const passwordInput = screen.getByPlaceholderText(/password/i);
    const toggleButton = document.querySelector('button[aria-label*="password"]');

    expect(passwordInput).toHaveAttribute('type', 'password');

    if (toggleButton) {
      await user.click(toggleButton);
      expect(passwordInput).toHaveAttribute('type', 'text');

      await user.click(toggleButton);
      expect(passwordInput).toHaveAttribute('type', 'password');
    }
  });

  it('handles keyboard navigation', async () => {
    render(<LoginForm />);

    const emailInput = screen.getByPlaceholderText(/email address/i);
    const passwordInput = screen.getByPlaceholderText(/password/i);
    const submitButton = screen.getByTestId('submit-button');

    // Enter key submission
    fireEvent.keyDown(submitButton, { key: 'Enter' });
    // The form should be submitted but mockLogin won't be called directly due to form validation
    // We'll just check that the event doesn't throw an error
    expect(true).toBe(true);
  });

  it('is accessible', () => {
    const { container } = render(<LoginForm />);

    // Check for proper ARIA attributes
    const form = container.querySelector('form');
    expect(form).toHaveAttribute('role', 'form');

    const emailInput = screen.getByPlaceholderText(/email address/i);
    expect(emailInput).toHaveAttribute('type', 'email');
    expect(emailInput).toHaveAttribute('required');

    const passwordInput = screen.getByPlaceholderText(/password/i);
    expect(passwordInput).toHaveAttribute('type', 'password');
    expect(passwordInput).toHaveAttribute('required');

    const submitButton = screen.getByTestId('submit-button');
    expect(submitButton).toHaveAttribute('type', 'submit');
  });

  it('matches snapshot', () => {
    const { container } = render(<LoginForm />);
    expect(container).toMatchSnapshot();
  });
});