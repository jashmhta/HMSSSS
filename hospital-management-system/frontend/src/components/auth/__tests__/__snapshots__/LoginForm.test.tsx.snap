// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm matches snapshot 1`] = `
<div>
  <div
    class="min-h-screen flex items-center justify-center bg-gray-50 py-12 px-4 sm:px-6 lg:px-8"
  >
    <div
      class="max-w-md w-full space-y-8"
    >
      <div>
        <h2
          class="mt-6 text-center text-3xl font-extrabold text-gray-900"
        >
          Sign in to your account
        </h2>
      </div>
      <form
        class="mt-8 space-y-6"
        role="form"
      >
        <div
          class="rounded-md shadow-sm -space-y-px"
        >
          <div>
            <label
              class="sr-only"
              for="email"
            >
              Email address
            </label>
            <input
              autocomplete="email"
              class="appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 focus:z-10 sm:text-sm"
              id="email"
              name="email"
              placeholder="Email address"
              required=""
              type="email"
              value=""
            />
          </div>
          <div
            class="relative"
          >
            <label
              class="sr-only"
              for="password"
            >
              Password
            </label>
            <input
              autocomplete="current-password"
              class="appearance-none rounded-none relative block w-full px-3 py-2 pr-10 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-b-md focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 focus:z-10 sm:text-sm"
              id="password"
              name="password"
              placeholder="Password"
              required=""
              type="password"
              value=""
            />
            <button
              aria-label="Show password"
              class="absolute inset-y-0 right-0 pr-3 flex items-center"
              type="button"
            >
              <svg
                class="h-5 w-5 text-gray-400"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
              >
                <path
                  d="M13.875 18.825A10.05 10.05 0 0112 19c-4.478 0-8.268-2.943-9.543-7a9.97 9.97 0 011.563-3.029m5.858.908a3 3 0 114.243 4.243M9.878 9.878l4.242 4.242M9.88 9.88l-3.29-3.29m7.532 7.532l3.29 3.29M3 3l3.59 3.59m0 0A9.953 9.953 0 0112 5c4.478 0 8.268 2.943 9.543 7a10.025 10.025 0 01-4.132 5.411m0 0L21 21"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="flex items-center justify-between"
        >
          <div
            class="flex items-center"
          >
            <input
              class="h-4 w-4 text-indigo-600 focus:ring-indigo-500 border-gray-300 rounded"
              id="remember-me"
              name="rememberMe"
              type="checkbox"
            />
            <label
              class="ml-2 block text-sm text-gray-900"
              for="remember-me"
            >
              Remember me
            </label>
          </div>
          <div
            class="text-sm"
          >
            <button
              class="font-medium text-indigo-600 hover:text-indigo-500"
              type="button"
            >
              Forgot your password?
            </button>
          </div>
        </div>
        <div>
          <button
            class="group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500 disabled:opacity-50 disabled:cursor-not-allowed"
            data-testid="submit-button"
            type="submit"
          >
            Sign in
          </button>
        </div>
        <div
          class="text-center"
        >
          <p
            class="text-sm text-gray-600"
          >
            Don't have an account?
             
            <button
              class="font-medium text-indigo-600 hover:text-indigo-500"
              type="button"
            >
              Sign up
            </button>
          </p>
        </div>
      </form>
      <div
        class="mt-8 border-t border-gray-200 pt-6"
      >
        <div
          class="text-center"
        >
          <h3
            class="text-sm font-medium text-gray-900 mb-2"
          >
            Demo Credentials:
          </h3>
          <div
            class="space-y-1 text-xs text-gray-600"
          >
            <p>
              admin@hms.com / admin123
            </p>
            <p>
              doctor@hms.com / doctor123
            </p>
            <p>
              patient@hms.com / patient123
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
